### DISCUSSION TITLE
Young's modulus calculation discrepancies

### DISCUSSION THREAD

**Discussion by lucapbelotti** [mtex-toolbox/mtex#1030] [Ask Anything] [answered]
Hi everyone,

Maybe it is a simple question, but I am struggling to understand the following:

I am calculating the stiffness tensor using the following :
```
CS = crystalSymmetry('m-3m');
% Single crystal elastic tensor
c11 = 204.6; c12 = 137.6; c44 = 126.2;
C = stiffnessTensor(...
  [[  c11   c12   c12    0   0   0];...
  [   c12   c11   c12    0   0   0];...
  [   c12   c12   c11    0   0   0];...
  [   0     0     0     c44  0   0];...
  [   0     0     0      0   c44 0];...
  [   0     0     0      0   0  c44]],CS);
% Examples
d = [Miller(1,0,0,CS); Miller(1,1,0,CS); Miller(1,1,1,CS)];
E_d = C.YoungsModulus(d)
E_spherical = C.YoungsModulus;
E_spherical_d = E_spherical.eval(d)
```
However, when I evaluate Young's modulus for different directions, I get different answers as follows:
```
E_d =   93.9410; 193.6754; 299.7562
E_spherical_d =  105.4305; 193.0946; 283.7204
```
I think it is because the `E_spherical = C.YoungsModulus` is a spherical function, but I am wondering if there is a way around this difference?

**Comment by kilir** [comment, marked as answer]
Hi,
the issue seems related to the very small bandwith used when evaluating the tensor.
Changing 
https://github.com/mtex-toolbox/mtex/blob/5caf324d85954fc47ccbb0fb4c1f7a7084b9c046/TensorAnalysis/%40tensor/directionalMagnitude.m#L20

to a bandwidth of e.g. 24 produces acceptable results for me.

Does that work?

Cheers,
Rüdiger

**Reply by lucapbelotti** [reply]
Rüdiger,

Yes, that works perfectly. Thanks

**Comment by ralfHielscher** [comment]
Hi Luca,

this is indeed a bug which we should fix as soon as possible. Thank you for reporting this.

Ralf.

**Reply by ralfHielscher** [reply]
It has now been fixed in be04ffd

**Reply by lucapbelotti** [reply]
Thanks Ralf.

Luca
